src/transports/smart.c

```diff
@@ -279,30 +279,34 @@ static int git_smart__connect(
 		return error;
 
 	/* Detect capabilities */
-	if (git_smart__detect_caps(first, &t->caps, &symrefs) < 0) {
-		free_symrefs(&symrefs);
-		return -1;
-	}
```

Comment:
I actually find this commit a bit hard to digest, as it simultaneously moves around code and changes it at the same time. I'd welcome it if you split it up into multiple commits to make it easier to understand what is going on here

Reply:
Sorry, this seemed a simple fix, then memory-management happened. I'll try to split (the crux of the change is that `git_smart__detect_caps` returns `GIT_ENOTFOUND` if it stumble on a weird pkt, which allows its caller to know if the `symrefs` it got are valid or not.

Reply:
Thanks :)

Reply:
Hopefully the commits are more separated now, which clarifies what is happening. Note that I erroneously said `symref` above, but my understanding of the issue is that it might (maliciously) happen that the remote sent back one ref (refs is a vector of git_pkt in this code 😮) but doesn't follow through, so this `strcmp` will segfault on a NULL.

Thinking back a bit more, I don't know if that's plausible protocol-wise, but at least coverity should be pleased…